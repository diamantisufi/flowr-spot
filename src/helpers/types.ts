export const ActiveModal = {
  LOGIN: 'login',
  LOGIN_SUCCESS: 'login_success',
  REGISTER: 'register',
  REGISTER_SUCCESS: 'register_success',
  PROFILE: 'profile',
  SETTINGS: 'settings',
}

export type FlowerListResponseData = {
  id: string
  name: string
  latin_name: string
  profile_picture: string
  favorite: boolean
  sightings: number
}

export type LoginRequestParams = {
  email: string
  password: string
}

export type RegisterRequestParams = {
  first_name: string
  last_name: string
  email: string
  password: string
  date_of_birth: string
}

export type RegisterResponseData = {
  auth_token: string
  error?: string
}

export type LoginResponseData = {
  auth_token: string
  error?: string
}
