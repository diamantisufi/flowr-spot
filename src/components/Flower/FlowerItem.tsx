import React from "react";

export const FlowerItem: React.FC = () => {
  return <div>FlowerItem</div>;
};
