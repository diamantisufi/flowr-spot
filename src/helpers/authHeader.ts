import { useContext } from 'react'
import { store } from 'store'
import { User } from './types'

export const authHeader = () => {
  let token = localStorage.getItem('auth_token')

  if (token !== null) {
    return { Authorization: 'Bearer ' + token }
  } else {
    return {}
  }
}

interface AppState {
  user: User
  activeModal: string
  auth: {
    token: string
  }
}

export const IsAuthenticated = () => {
  // token exp was not required

  const context = useContext(store)
  const state = context?.state as AppState
  const isAuthenticated = !!state?.auth?.token

  return isAuthenticated
}
