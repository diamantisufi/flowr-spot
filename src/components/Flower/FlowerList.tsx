import React from 'react'
import { FlowerItem } from './FlowerItem'
import { FlowerListResponseData } from 'helpers/types'

type FlowerListProps = {
  data: FlowerListResponseData[]
}

export const FlowerList: React.FC<FlowerListProps> = ({ data }) => {
  return (
    <div className='grid grid-cols-1 md:grid-cols-3 lg:grid-cols-4 gap-5 p-4 mt-16'>
      {data?.map((flower, index) => (
        <FlowerItem key={flower?.id} flower={flower} />
      ))}
    </div>
  )
}
