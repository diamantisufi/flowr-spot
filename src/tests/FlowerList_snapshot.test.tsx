import React from 'react'
import { render } from '@testing-library/react'
import { FlowerList } from 'components/Flower/FlowerList'
import { FlowerListResponseData } from 'helpers/types'

describe('FlowerList', () => {
  it('matches snapshot', () => {
    const mockFlowers: FlowerListResponseData[] = [
      {
        id: '1',
        name: 'Rose',
        latin_name: 'Rosa',
        profile_picture: 'rose.jpg',
        sightings: 10,
        favorite: true,
      },
      {
        id: '2',
        name: 'Sunflower',
        latin_name: 'Helianthus',
        profile_picture: 'sunflower.jpg',
        sightings: 15,
        favorite: false,
      },
      // Add more mock flower data as needed...
    ]

    const { asFragment } = render(<FlowerList data={mockFlowers} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
