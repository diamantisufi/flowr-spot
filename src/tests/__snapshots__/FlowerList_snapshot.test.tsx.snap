// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowerList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 my-16 mx-2"
    id="flowers"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-3 lg:grid-cols-4 gap-5"
    >
      <div
        class="relative h-[22rem]"
      >
        <img
          alt="Rose"
          class="w-full h-full object-cover object-center rounded"
          src="rose.jpg"
        />
        <div
          class="absolute inset-0 flowerItemCard"
        >
          <div
            class="absolute bottom-0 left-0 right-0 bg-opacity-50 flex flex-col justify-center text-center font-ubuntu text-white"
          >
            <h2
              class="text-xl"
            >
              Rose
            </h2>
            <p
              class="text-xs italic mb-5leading-3 opacity-70"
            >
              Rosa
            </p>
            <div
              class="w-full flex items-center justify-center"
            >
              <button
                class="text-white text-center font-ubuntu text-xs leading-3 my-5 bg-[#00000070] rounded-full w-1/2 py-3 px-4 false"
              >
                <p
                  class="text-gray-300"
                >
                  10 sightings
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative h-[22rem]"
      >
        <img
          alt="Sunflower"
          class="w-full h-full object-cover object-center rounded"
          src="sunflower.jpg"
        />
        <div
          class="absolute inset-0 flowerItemCard"
        >
          <div
            class="absolute bottom-0 left-0 right-0 bg-opacity-50 flex flex-col justify-center text-center font-ubuntu text-white"
          >
            <h2
              class="text-xl"
            >
              Sunflower
            </h2>
            <p
              class="text-xs italic mb-5leading-3 opacity-70"
            >
              Helianthus
            </p>
            <div
              class="w-full flex items-center justify-center"
            >
              <button
                class="text-white text-center font-ubuntu text-xs leading-3 my-5 bg-[#00000070] rounded-full w-1/2 py-3 px-4 false"
              >
                <p
                  class="text-gray-300"
                >
                  15 sightings
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
